backend/test/infra/controller/GroupController.test.ts

```diff
@@ -0,0 +1,182 @@
+import Fastify, { type FastifyInstance } from "fastify";
+import { Group } from "src/core/entity/Group";
+import {
+  type CreateGroupRequestObject,
+  type CreateGroupResponseObject,
+} from "src/core/usecase/CreateGroup";
+import {
+  type RenameGroupRequestObject,
+  type RenameGroupResponseObject,
+} from "src/core/usecase/RenameGroup";
+import { GroupController } from "src/infra/controller/GroupController";
+import { GroupRepositoryMock } from "test/core/usecase/test-helpers";
+import { afterEach, beforeEach, describe, expect, test, vi } from "vitest";
+
+// Warning, using the absolute path doesn't work here
+vi.mock("../../../src/core/usecase/CreateGroup");
+vi.mock("../../../src/core/usecase/RenameGroup");
+
+describe("GroupController", () => {
+  describe("Create group", () => {
+    let server: FastifyInstance;
+    const groupRepo = new GroupRepositoryMock();
+    const createGroupMock = {
+      execute: vi.fn<CreateGroupRequestObject, CreateGroupResponseObject>(),
+    };
+
+    beforeEach(async () => {
+      const module = await import("src/core/usecase/CreateGroup");
+      // @ts-expect-error It isn't typed as a Mock
+      module.CreateGroup.mockReturnValue(createGroupMock);
+
+      // server = await createServer("test");
+      server = Fastify({ logger: false });
+      await server.register(GroupController, {
+        prefix: "/group",
+        repositories: { group: groupRepo },
+      });
+      await server.ready();
+    });
+
+    afterEach(() => {
+      vi.clearAllMocks();
+      vi.resetAllMocks();
+    });
+
+    test("Given a name, it should succeed with a status 201 and return the newly created group", async () => {
+      // Arrange
+      createGroupMock.execute.mockResolvedValue({
```

Comment:
Wouldn't it be better to create the new group beforehand and then use the newGroup.name throughout the test instead of the string "new group" , and do the same with the id ?